test/unit/test_keyring_raw_aes.py

```diff
@@ -0,0 +1,343 @@
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        wrapping_key=_WRAPPING_KEY,
+    )
+    assert test.key_name == _KEY_ID
+    assert test.key_namespace == _PROVIDER_ID
+    assert test._wrapping_algorithm == WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING
+    assert test._wrapping_key == _WRAPPING_KEY
+
+
+def test_missing_required_parameters():
+    with pytest.raises(Exception) as exc_info:
+        RawAESKeyring(
+            key_namespace=_PROVIDER_ID, wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING
+        )
+    assert exc_info.errisinstance(TypeError)
+
+
+def test_invalid_values_as_parameter():
+    with pytest.raises(Exception) as exc_info:
+        RawAESKeyring(
+            key_namespace=_PROVIDER_ID,
+            key_name=_KEY_ID,
+            wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+            wrapping_key=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        )
+    assert exc_info.errisinstance(TypeError)
+
+
+def test_on_encrypt_when_data_encryption_key_given(patch_generate_data_key):
+    test_raw_aes_keyring = RawAESKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        wrapping_key=_WRAPPING_KEY,
+    )
+
+    test_raw_aes_keyring.on_encrypt(encryption_materials=get_encryption_materials_with_data_encryption_key())
+    # Check if keyring is generated
+    assert not patch_generate_data_key.called
+
+
+def test_keyring_trace_on_encrypt_when_data_encryption_key_given():
+    test_raw_aes_keyring = RawAESKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        wrapping_key=_WRAPPING_KEY,
+    )
+
+    test = test_raw_aes_keyring.on_encrypt(encryption_materials=get_encryption_materials_with_data_encryption_key())
+
+    for keyring_trace in test.keyring_trace:
+        if keyring_trace.wrapping_key.key_info == _KEY_ID:
+            # Check keyring trace does not contain KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY
+            assert KeyringTraceFlag.WRAPPING_KEY_GENERATED_DATA_KEY not in keyring_trace.flags
+
+
+def test_on_encrypt_when_data_encryption_key_not_given():
+
+    test_raw_aes_keyring = RawAESKeyring(
+        key_namespace=_PROVIDER_ID,
+        key_name=_KEY_ID,
+        wrapping_algorithm=WrappingAlgorithm.AES_256_GCM_IV12_TAG16_NO_PADDING,
+        wrapping_key=_WRAPPING_KEY,
+    )
+
+    original_number_of_encrypted_data_keys = len(
+        get_encryption_materials_without_data_encryption_key().encrypted_data_keys
+    )
+
+    test = test_raw_aes_keyring.on_encrypt(encryption_materials=get_encryption_materials_without_data_encryption_key())
+
+    # Check if data key is generated
+    assert test.data_encryption_key and test.data_encryption_key is not None
```

Comment:
This value being truthy *but not `None`* doesn't really tell us anything. There is no valid falsy value for it other than `None`, and I worry that putting a check like this might lead a future reader to think that there is.
```suggestion
    assert test.data_encryption_key is not None
```